signature/src/signer.rs

```diff
@@ -24,6 +24,18 @@ pub trait Signer<S> {
     fn try_sign(&self, msg: &[u8]) -> Result<S, Error>;
 }
 
+/// Equivalent of [`Signer`] but the message is provided in non-contiguous byte slices.
+pub trait MultiPartSigner<S> {
+    /// See [`Signer::sign()`].
+    fn multi_part_sign(&self, msg: &[&[u8]]) -> S {
```

Comment:
Small nit, but "multipart" is typically a closed-compound word, IMO, in which case the Rust style would be:

```suggestion
pub trait MultipartSigner<S> {
    /// See [`Signer::sign()`].
    fn multipart_sign(&self, msg: &[&[u8]]) -> S {
```

Reply:
The comment could also be better as well, IMO

Reply:
Done.

Appropriately renamed method names as well.